sdk/identity/azure-identity/azure/identity/_credentials/client_secret.py

```diff
@@ -24,10 +20,19 @@ class ClientSecretCredential(ClientCredentialBase):
     :keyword cache_persistence_options: configuration for persistent token caching. If unspecified, the credential
         will cache tokens in memory.
     :paramtype cache_persistence_options: ~azure.identity.TokenCachePersistenceOptions
+    :keyword bool allow_broker: Brokers provide single sign-on, device identification, and application identification
+        verification. If this parameter is set to True, the broker will be used when possible. Defaults to False.
     """
 
-    def __init__(self, tenant_id, client_id, client_secret, **kwargs):
-        # type: (str, str, str, **Any) -> None
+    def __init__(
+            self,
+            tenant_id: str,
+            client_id: str,
+            client_secret: str,
+            *,
+            allow_broker: bool = False,
```

Comment:
Is `allow_broker` a term used in other libraries (`msal`, Identity in other languages, etc.)? Has the team agreed on this, or do you think something like `use_broker` would make more sense? I'm not familiar with the feature, so I'll defer to you though 🙂

Reply:
It is used in msal.

This is a beta feature and the parameter name is subject to change when doing arch board review. :) (btw, it is unlikely to GA in copper)

Reply:
Got it, thanks!